### PR TITLE
Fail on apply if DVU roots are not empty

### PR DESCRIPTION
**Pull request by nickgerace**
## Description

This PR ensures that we fail on applying a change set in `sdf` if DVU roots are not empty. A toast will be fired if this happens, which should be rare and require no additional user action.

### CODE REVIEW

File: lib/sdf-server/src/service/change_set.rs [hunk 1]
```diff
@@ -73,11 +75,16 @@ pub type ChangeSetResult<T> = std::result::Result<T, ChangeSetError>;
...
         let (status, error_message) = match self {
-            ChangeSetError::ChangeSetNotFound => (StatusCode::NOT_FOUND, self.to_string()),
-            ChangeSetError::DalChangeSetApply(_) => (StatusCode::CONFLICT, self.to_string()),
             ChangeSetError::ActionAlreadyEnqueued(_) => {
                 (StatusCode::NOT_MODIFIED, self.to_string())
             }
+            ChangeSetError::ChangeSetNotFound => (StatusCode::NOT_FOUND, self.to_string()),
+            ChangeSetError::DalChangeSetApply(_) => (StatusCode::CONFLICT, self.to_string()),
+            ChangeSetError::DvuRootsNotEmpty(_) => (
+                StatusCode::PRECONDITION_REQUIRED,
```
Comment (johnrwatson) [inline lib/sdf-server/src/service/change_set.rs:84]: super nice

Comment (nickgerace) [reply lib/sdf-server/src/service/change_set.rs:84]: Thank you Mr. Watson 🫡

File: app/web/src/components/toasts/RetryApply.vue
Comment (nickgerace) [inline app/web/src/components/toasts/RetryApply.vue:1]: This is a direct fork of `FiveHundredError`. For this PR, I did not want to bundle both errors into one Vue component, but I could see a future where we have a `GenericError` toast with an optional title (and if one isn't provided, you get the error title that `FiveHundredError` currently has).

cc @wendybujalski

Comment (wendybujalski) [reply app/web/src/components/toasts/RetryApply.vue:1]: Yeah, unifying into a GenericError component will definitely be good to do down the line. But this works for now!

**Review by wendybujalski — APPROVED** [review]
👍

File: app/web/src/components/toasts/RetryApply.vue
Comment (nickgerace) [reply app/web/src/components/toasts/RetryApply.vue:1]: Thank you @wendybujalski!

**Pull request merged**